src/main/kotlin/service/user/balance/BalanceIncreaseRateManager.kt

```diff
@@ -0,0 +1,48 @@
+package service.user.balance
+
+import model.User
+import model.Users
+import model.getPokemons
+import utility.RedisFactory
+
+class BalanceIncreaseRateManager(val user: User) {
+    fun increaseBalanceBasedOnIncreaseRate(balanceManager: BalanceManager) {
+        val currentIncreaseRatePerMinute = retrieveIncreaseRate()
+        if (currentIncreaseRatePerMinute != null) balanceManager.increaseCurrentBalance(currentIncreaseRatePerMinute)
```

Comment:
So what if it is null? How can that be? Shouldn't that be an exception? Or make the method more descriptive.

Reply:
Actually you are right! :D